### PR TITLE
Storybook snapshots - introduce visual tests for React components

### LINKED ISSUE #88: Do some research about snapshot tests using Storybook.
Research about snapshot test and provide report about possibilities

### PR DESCRIPTION
**Pull request by SzajekSzymon**
## Description
Fixes #88.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,12 +10,14 @@
     "dev": "webpack",
     "watch": "webpack --watch",
     "test": "jest -c jest.config.js",
+    "test:snapshot": "jest -c jest-snapshot.config.js",
+    "snapshot": "start-server-and-test storybook http-get://localhost:6006 test:snapshot",
```
Comment (Voycawojka) [inline package.json:14]: What's the difference between `test:snapshot` and `snapshot`?

File: src/js/snapshots/imageSnapshot.snapshot.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    await page.goto('http://localhost:6006/iframe.html?id=logic-components-sidepanel--sidepanel');
+    await page.evaluateHandle('document.fonts.ready');
+    const image = await page.screenshot();
+
+    expect(image).toMatchImageSnapshot(config);
+  });
+});
+
+describe('FragmentGantt', () => {
+  it('FragmentGantt visually looks correct', async () => {
```
Comment (Voycawojka) [inline src/js/snapshots/imageSnapshot.snapshot.js:38]: Not a big issue but wording seems a bit jarring to me. How about `is visually correct`, `looks correct` or `matches its snapshot`?

File: src/js/snapshots/markupSnapshot.spec.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import initStoryshots, { snapshotWithOptions } from '@storybook/addon-storyshots';
+
+initStoryshots({
+  test: snapshotWithOptions({
+    // eslint-disable-next-line consistent-return
```
Comment (Voycawojka) [inline src/js/snapshots/markupSnapshot.spec.js:21]: I wonder if we should tweak our eslint configuration? 🤔  There's a lot of `eslint-disable` comments in the code base and what's the point of eslint then? 
I guess one could argue it indicates the developer is aware of the possible issue

File: src/js/snapshots/imageSnapshot.snapshot.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-undef */
+
+// page object is from jest-puppeteer API
+
+const config = {
```
Comment (xolir) [inline src/js/snapshots/imageSnapshot.snapshot.js:21]: You can configure toMatchImageSnapshot before tests are happening in setup file, that way you do not need to keep that in test file nor add that to every method call

File: src/js/snapshots/imageSnapshot.snapshot.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    await page.evaluateHandle('document.fonts.ready');
+    const image = await page.screenshot();
+
+    expect(image).toMatchImageSnapshot(config);
+  });
+});
+
+describe('FragmentGantt', () => {
+  it('is visually correct', async () => {
+    await page.goto('http://localhost:6006/iframe.html?id=logic-components-sidepanel-fragmentgantt--fragmentgantt');
```
Comment (xolir) [inline src/js/snapshots/imageSnapshot.snapshot.js:39]: All of them are fairly similar, can this be extracted to helper instead that will take URL and return test?

File: src/js/snapshots/imageSnapshot.snapshot.js (hunk 4 above)
Comment (SzajekSzymon) [reply src/js/snapshots/imageSnapshot.snapshot.js]: fixed

File: src/js/snapshots/imageSnapshot.snapshot.js (hunk 5 above)
Comment (SzajekSzymon) [reply src/js/snapshots/imageSnapshot.snapshot.js]: fixed

File: package.json (hunk 1 above)
Comment (SzajekSzymon) [reply package.json:14]: I don't know why, but when i use this script:
```
 "snapshot": "start-server-and-test storybook http-get://localhost:6006 jest -c jest-snapshot.config.js",
```

snapshot command fire snapshots and all of unit tests

**Pull request merged**